node-deb

```diff
@@ -683,25 +683,39 @@ find "$@" -type f -print0 | {
   done
 }
 
-if [ -d "$source_dir/node_modules" ]; then
-  npm ls --parseable --prod | sed -e "s/$(escape "$source_dir")//g" | grep -Ev '^$' | sed -e 's:/node_modules/::g' | {
-    while IFS="\n" read -r -d '' dir; do
-      log_debug "Copying dir: $dir"
-      cp -rf "$source_dir/node_modules/$dir" "$deb_dir/usr/share/$package_name/app/node_modules/"
-    done
-  }
-fi
+app_dir="$deb_dir/usr/share/$package_name/app"
 
-if ! [ -f "$deb_dir/usr/share/$package_name/app/package.json" ]; then
+if ! [ -f "$app_dir/package.json" ]; then
   log_info "Including 'package.json' in the Debian package."
   cp -pf './package.json' "$deb_dir/usr/share/$package_name/app/"
 fi
 
-if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$deb_dir/usr/share/$package_name/app/npm-shrinkwrap.json" ]; then
+if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$app_dir/npm-shrinkwrap.json" ]; then
   log_info "Including 'npm-shrinkwrap.json' in the Debian package."
-  cp -pf './npm-shrinkwrap.json' "$deb_dir/usr/share/$package_name/app/"
+  cp -pf './npm-shrinkwrap.json' "$app_dir"
+fi
+
+if [ -f './yarn.lock' ] && ! [ -f "$app_dir/yarn.lock" ]; then
+  log_info "Including 'yarn.lock' in the Debian package."
+  cp -pf './yarn.lock' "$app_dir"
 fi
 
+current_dir="$(pwd)"
+cd "$app_dir"
+if [ -f "./yarn.lock" ]; then
+  log_info "Building node_modules dir with yarn"
+  yarn install --no-progress --production 2>&1 > /dev/null | sed "s/^/YARN: /"
+else
+  log_info "Building node_modules dir with npm"
+  npm install --no-progress --production  2>&1 > /dev/null | sed "s/^/NPM:  /"
+fi
+cd "$current_dir"
+
+# some packages might store modules with u+rw only which
+# will be a problem if the application is executed as user
+# but the files are owned by root
+find "$app_dir/node_modules" -print0 | xargs -0 chmod go+rX
```

Comment:
I think this is better solved by changing the `postinst` script to transfer ownership of the files to `user:group`. In fact, I think there was a regression as I think this used to be how the scripts worked,